spk/cops/src/service-setup.sh

```diff
@@ -0,0 +1,122 @@
+DSM6_WEB_DIR="/var/services/web"
+if [ "${SYNOPKG_DSM_VERSION_MAJOR}" -ge 7 ]; then
+   WEB_DIR="/var/services/web_packages"
+else
+   WEB_DIR="${DSM6_WEB_DIR}"
+fi
+CFG_FILE="${WEB_DIR}/${SYNOPKG_PKGNAME}/${CFG_FILE_NAME}"
+BUILDNUMBER="$(/bin/get_key_value /etc.defaults/VERSION buildnumber)"
+
+USER="$([ "${BUILDNUMBER}" -ge "4418" ] && echo -n http || echo -n nobody)"
+PHP_CONFIG_LOCATION="$([ "${BUILDNUMBER}" -ge "7135" ] && echo -n /usr/local/etc/php56/conf.d || echo -n /etc/php/conf.d)"
+SC_GROUP="http"
+
+service_preinst ()
+{
+    if [ "${SYNOPKG_PKG_STATUS}" == "INSTALL" ]; then
+        if [ ! -f "${wizard_calibre_dir}/metadata.db" ]; then
+            echo "Metadata.db cannot be found. Please verify that the Calibre directory was entered correctly."
+            exit 1
+        fi
+    fi
+}
+
+service_postinst ()
+{
+      if [ "${SYNOPKG_DSM_VERSION_MAJOR}" -lt 7 ]; then
+    
+        # Install the web interface
+        cp -pR "${SYNOPKG_PKGDEST}/share/${SYNOPKG_PKGNAME}" "${WEB_DIR}"
+    
+        # Configure open_basedir
+        if [ "${USER}" == "nobody" ]; then
+            echo -e "<Directory \"${WEB_DIR}/${SYNOPKG_PKGNAME}\">\nphp_admin_value open_basedir none\n</Directory>" > /usr/syno/etc/sites-enabled-user/${SYNOPKG_PKGNAME}.conf
+        else
+            echo -e "[PATH=${WEB_DIR}/${SYNOPKG_PKGNAME}]\nopen_basedir = Null" > ${PHP_CONFIG_LOCATION}/${PACKAGE_NAME}.ini
+        fi
+      fi
+
+    if [ "${SYNOPKG_PKG_STATUS}" == "INSTALL" ]; then
+        # Create a default configuration file
+        if [ ! -f "${CFG_FILE}" ]; then
+          cp "${DEFAULT_CFG_FILE}" "${CFG_FILE}"
+          url_rewriting=$([ "${wizard_use_url_rewriting}" == "true" ] && echo "1" || echo "0")
+          sed -i -e "s|@calibre_dir@|${wizard_calibre_dir:=/volume1/calibre/}|g" ${CFG_FILE}
+          sed -i -e "s|@cops_title@|${wizard_cops_title:=COPS}|g" ${CFG_FILE}
+          sed -i -e "s|@use_url_rewriting@|${url_rewriting:=0}|g" ${CFG_FILE}
+          chmod ga+w "${CFG_FILE}"
+        fi
+
+      if [ "${SYNOPKG_DSM_VERSION_MAJOR}" -lt 7 ]; then
+
+        # Set permissions
+        if [ "${BUILDNUMBER}" -ge "4458" ];  then
+            # DSM5+
+            # Set permissions on directory structure
+            set_syno_permissions "${wizard_calibre_dir}"
+            # Set permissions on metadata.db
+            if [ ! "`synoacltool -get "${wizard_calibre_dir}/metadata.db"| grep "group:${SC_GROUP}:allow:rwxpdDaARWc."`" ]; then
+                synoacltool -add "${wizard_calibre_dir}/metadata.db" "group:${SC_GROUP}:allow:rwxpdDaARWc:----" > /dev/null 2>&1
+            fi
+        else
+            #DSM4
+            chown ${USER} ${wizard_calibre_dir:=/volume1/calibre/}
+            chmod u+rw ${wizard_calibre_dir:=/volume1/calibre/}
+            chown ${USER} ${wizard_calibre_dir:=/volume1/calibre/}/metadata.db
+            chmod u+rw ${wizard_calibre_dir:=/volume1/calibre/}/metadata.db            
```

Comment:
Is this still relevant?

Reply:
Good point. Given the comment I guess it can be removed safely.

Reply:
So I removed everything which sounded like DSM 4 specifics.

The only thing is that I don't own a DSM 6 to validate it still works on those :/

Reply:
I got one, let me know the specifics you'd need testing on and I'll check it out.

Reply:
Here's a plain default install result (e.g. I've only hit next, no kobo selection):
![Capture d’écran du 2022-05-01 07-33-40](https://user-images.githubusercontent.com/17141866/166144023-8e9e933e-f5e5-48ec-b608-209415027c5c.png)

Reply:
Have you checked the path specified ? Does it match an existing directory ?

Reply:
Regarding the PHP configuration I guess I should be documenting it and adding this as part of the DSM 6 wizard specifics. (It's a matter of enabling the right extensions)

Reply:
> Have you checked the path specified ? Does it match an existing directory ?

Well there is no `/volume1/calibre` directory... I presumed (perhaps wrongly) that it was going to generate one if non-existant...

Reply:
Actually it's supposed to be the path to an existing calibre directory... This may sound awkward but, as of now, it doesn't create any folder structure.

Just updated the wizard to explicitely mention the list of extensions to enable.

Reply:
> Actually it's supposed to be the path to an existing calibre directory... 

I haven't noticed if this was clearly mentionned.  Couldn't it generate an "empty" db file in case it's non-existant?  On the other hand, I don't know exactly what this app is for nor how to use it so take this comment with a grain of salt.

Otherwise, if you want me to test that out could you provide me with an default db file that I could confirm testing with on DSM6?  Perhaps that "empty" db file could actually be provided with the package... just a thought.

Reply:
> > Actually it's supposed to be the path to an existing calibre directory...
> 
> I haven't noticed if this was clearly mentionned. Couldn't it generate an "empty" db file in case it's non-existant? On the other hand, I don't know exactly what this app is for nor how to use it so take this comment with a grain of salt.

The idea of this app is to come as a companion to an existing calibre installation which would be handling the writing and organizing part. cops is "merely" a browser component which integrates smoothly with web browsers and is compliant with other ebook readers (for example Kobo ones, and it happens I use one :) ).

I believe we could generate an empty DB but I'm not sure of the added value here (nor that it would be maintainable). Will be checking.

> 
> Otherwise, if you want me to test that out could you provide me with an default db file that I could confirm testing with on DSM6? Perhaps that "empty" db file could actually be provided with the package... just a thought.

That's for sure. I'll attach one to the PR right away.

Reply:
Did a git pull and rebuilt the package.  It's nice to mention which PHP extensions are needed but having a URL to point for help would be beneficial as otherwise the user will ask himself : am I missing something?  We could probably create a WebStation wiki page on our spksrc github and include our finding for creating packages and al (just a thought).

Additionally, perhaps you should include an additional `SPK_DEPENDS` for php?

An error occured while changing permissions:
```
2022/05/01 15:58:31	===> Step postinst. USER=http GROUP= SHARE_PATH=
2022/05/01 15:58:31	Begin save_wizard_variables
2022/05/01 15:58:31	End save_wizard_variables
2022/05/01 15:58:31	Begin service_postinst
2022/05/01 15:58:31	ACL version: 1 
2022/05/01 15:58:31	Archive: has_ACL,is_support_ACL 
2022/05/01 15:58:31	Owner: [th0ma7(user)] 
2022/05/01 15:58:31	--------------------- 
2022/05/01 15:58:31		 [0] group:administrators:allow:rwxpdDaARWc--:fd--  (level:0)
2022/05/01 15:58:31	Granting '' group permissions on /volume1/calibre
2022/05/01 15:58:31	(synoacltool.c, 125)No such Group 
2022/05/01 15:58:31	Granting '' group basic permissions on /volume1/calibre
2022/05/01 15:58:31	(synoacltool.c, 125)No such Group 
2022/05/01 15:58:32	End service_postinst
2022/05/01 15:58:32	install cops 1.1.3-6 End postinst ret=[0]
2022/05/01 15:58:32	install cops 1.1.3-6 Begin /bin/rm -rf /volume1/@tmp/pkginstall
2022/05/01 15:58:32	install cops 1.1.3-6 End /bin/rm -rf /volume1/@tmp/pkginstall ret=[0]
2022/05/01 15:58:36	install cops 1.1.3-6 Begin start-stop-status start
2022/05/01 15:58:36	install cops 1.1.3-6 End start-stop-status start ret=[0]
```
This leads to the same output as previously (e.g. `Database error`) as it can't find the db file, and I assume that's the reason why.

Reply:
Re-tested but this time I made sure to create a `/volume1/calibre` folder using the GUI and that ACL was enabled.  Errors are now different which makes me think you would be better off adding a warning that ACL must be turned on.

Errors:
```
2022/05/03 10:05:24	===> Step postinst. USER=http GROUP= SHARE_PATH=
2022/05/03 10:05:24	Begin save_wizard_variables
2022/05/03 10:05:24	End save_wizard_variables
2022/05/03 10:05:24	Begin service_postinst
2022/05/03 10:05:24	Granting '' group permissions on /volume1/calibre
2022/05/03 10:05:24	(synoacltool.c, 125)No such Group 
2022/05/03 10:05:24	Granting '' group basic permissions on /volume1/calibre
2022/05/03 10:05:24	(synoacltool.c, 125)No such Group 
2022/05/03 10:05:24	End service_postins
```

Reply:
Actualy this is related to another issue (which I will be fixing).

But agree on the permissions message to add !

Reply:
Added permissions related message + fixed the GROUP logic to not have the failures reported